tests/phpunit/integration/Core/Authentication/AuthenticationTest.php

```diff
@@ -442,6 +448,144 @@ public function test_googlesitekit_connect() {
 		}
 	}
 
+	public function test_get_proxy_connect_user_url() {
+		$class  = new \ReflectionClass( Authentication::class );
+		$method = $class->getMethod( 'get_proxy_connect_user_url' );
+		$method->setAccessible( true );
+
+		$url = $method->invokeArgs(
+			new Authentication( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ),
+			array()
+		);
+
+		$this->assertNotEmpty( filter_var( $url, FILTER_VALIDATE_URL ) );
+		$this->assertStringStartsWith( admin_url(), $url );
+
+		$args = array();
+		parse_str( wp_parse_url( $url, PHP_URL_QUERY ), $args );
+
+		$this->assertArrayHasKey( 'action', $args );
+		$this->assertArrayHasKey( 'nonce', $args );
+
+		$this->assertEquals( Google_Proxy::ACTION_CONNECT_USER, $args['action'] );
+	}
+
+	public function test_set_connected_proxy_url() {
+		remove_all_actions( 'googlesitekit_authorize_user' );
+
+		$context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$options = new Options( $context );
+
+		$authentication = new Authentication( $context, $options );
+		$authentication->register();
+
+		$home_url_hook = function() {
+			return 'https://example.com/subsite/';
+		};
+
+		add_filter( 'home_url', $home_url_hook );
+		do_action( 'googlesitekit_authorize_user' );
+		remove_filter( 'home_url', $home_url_hook );
+
+		$this->assertEquals( 'example.com/subsite/', $options->get( Connected_Proxy_URL::OPTION ) );
+	}
+
+	public function test_check_connected_proxy_url() {
+		remove_all_actions( 'admin_init' );
+
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$context           = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$options           = new Options( $context );
+		$user_options      = new User_Options( $context );
+		$encrypted_options = new Encrypted_Options( $options );
+
+		$authentication = new Authentication( $context, $options, $user_options );
+		$authentication->register();
+
+		// Set connected proxy URL to something else to emulate URL mismatch.
+		$options->set( Connected_Proxy_URL::OPTION, '/' );
+
+		// Emulate credentials.
+		$encrypted_options->set(
+			Credentials::OPTION,
+			array(
+				'oauth2_client_id'     => 'xxx.apps.sitekit.withgoogle.com',
+				'oauth2_client_secret' => 'xxx-xxxx-xxxxx',
+			)
+		);
```

Comment:
I think we can use `fake_proxy_site_connection` from the `Fake_Site_Connection_Trait` for this instead.

Reply:
Updated.